cmd/osctl/cmd/logs.go

```diff
@@ -56,6 +59,23 @@ var logsCmd = &cobra.Command{
 					helpers.Fatalf("error streaming logs: %s", err)
 				}
 
+				if data.Metadata != nil && data.Metadata.Hostname != lastNode {
+					if lastNode != "" {
+						_, err = os.Stdout.Write([]byte("\n"))
+						helpers.Should(err)
+					}
+
+					lastNode = data.Metadata.Hostname
+
+					_, err = os.Stdout.Write([]byte(fmt.Sprintf("==> %s <==\n", lastNode)))
```

Comment:
Just an idea, but what if we were to prefix each line with the node name? I'm doing something similar here: https://github.com/talos-systems/talos-controller-manager/pull/1/files#diff-715c608d6874101ccd67a28d1a3b0020R360-R386

Reply:
I don't know which one is more readable, but I discovered one bug, looks like we need to accumulate buffers to get full lines, as output might be broken like this one:

```
2019/12/02 20:52:25.519329 log.go:87: Canceled [/machine.Machine/Logs] 4.068658992s stream rpc error: code = Canceled desc = context canceled (:authority=10.5.0.3:50000;content-type=application/grpc;prox
==> 10.5.0.2 <==
port is closing
	* error sending error back: rpc error: code = Unavailable desc = transport is closing

 (:authority=127.0.0.1;content-type=application/grpc;targets=10.5.0.2,10.5.0.3;user-agent=grpc-go/1.25.1)
2019/12/02 20:52:25.519239 log.go:87: Canceled [/machine.Machine/Logs] 4.068666633s stream rpc error: code = Canceled desc = context canceled (:authority=10.5.0.2:50000;content-type=application/grpc;proxyfrom=127.0.0.1;user-agent=grpc-go/1.25.1)
2019/12/02 21:03:51.382742 log.go:87: OK [/machine.Machine/ServiceList] 10.297594ms stream Success (:authority=127.0.0.1;content-type=application/grpc;user-agent=grpc-go/1.25.1)
2019/12/02 21:04:18.717108 log.go:87: Unknown [/machine.Machine/Logs] 3.908834ms stream rpc error: code = Unknown desc = open /run/system/log/machined.log: no such file or directory (:authority=127.0.0.1;content-type=application/grpc;user-agent=grpc-go/1.25.1)

==> 10.5.0.3 <==
yfrom=127.0.0.1;user-agent=grpc-go/1.25.1)
```

Reply:
heh, looks like this issue with lines is really complicated one, I tried two approaches, but I don't like both, let's put this on hold until I get some solid solution.

Reply:
got this addressed, code isn't that simple, but seems to work just fine

Reply:
I haven't updated the output format, but open to any format (node-prefixed, or like tail)

Reply:
+1 for something like `host file text`